src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -444,18 +513,19 @@ private InventoryEntryDraft withSupplyChannel(@Nonnull final InventoryEntryDraft
 
     /**
      * Method tries to create supply channel of given {@code supplyChannelKey} in CTP project.
-     * If operation succeed then {@link CompletionStage} with created {@link Channel} is returned, otherwise
-     * error callback function is executed and {@link CompletionStage} with {@link SyncProblemException} is returned.
+     * If operation succeed then {@link CompletionStage} containing {@link Optional} with created {@link Channel} is
+     * returned, otherwise error callback function is executed and {@link CompletionStage} with empty {@link Optional}
+     * is returned.
      *
      * @param supplyChannelKey key of supply channel that seems to not exists in a system
-     * @return {@link CompletionStage} instance that contains either created {@link Channel} instance when succeed
-     *      or {@link SyncProblemException} instance otherwise
+     * @return {@link CompletionStage} instance with {@link Optional} that may contain created {@link Channel}
      */
-    private CompletionStage<Channel> createMissingSupplyChannel(@Nonnull final String supplyChannelKey) {
+    private CompletionStage<Optional<Channel>> createMissingSupplyChannel(@Nonnull final String supplyChannelKey) {
```

Comment:
this method is far away of the group of methods that it's logically connected to